src/socket/dhcpv4.rs

```diff
@@ -256,6 +323,20 @@ impl Socket {
             dhcp_repr
         );
 
+        if let (Some(bootfile_name), Some((bootfile_name_len, bootfile_name_buffer))) =
+            (dhcp_repr.boot_file, self.bootfile_name_buffer.as_mut())
+        {
+            let len = bootfile_name.len().min(bootfile_name_buffer.len());
+            bootfile_name_buffer[..len].copy_from_slice(&bootfile_name.as_bytes()[..len]);
+            *bootfile_name_len = len;
+        }
+
+        if let (Some(inbox_options), Some(received_options)) =
+            (self.inbox_options.as_mut(), dhcp_repr.options)
+        {
+            let _ = inbox_options.emit(received_options.parse());
```

Comment:
this could be a memcpy instead of parse+emit

Reply:
I don't think it could be, since we need to emit as many options that will fit without truncating them.